subprojects/base-services/src/test/groovy/org/gradle/internal/exceptions/DefaultMultiCauseExceptionTest.groovy

```diff
@@ -124,8 +123,9 @@ class DefaultMultiCauseExceptionTest extends Specification {
         def failure = new TestMultiCauseException("message", [cause1, cause2])
 
         when:
-        def bytes = GUtil.serialize(failure)
-        def result = new ObjectInputStream(new ByteArrayInputStream(bytes)).readObject()
+        def baos = new ByteArrayOutputStream()
+        new ObjectOutputStream(baos).writeObject(failure)
+        def result = new ObjectInputStream(new ByteArrayInputStream(baos.toByteArray())).readObject()
```

Comment:
Is this test still valuable at this point?  We're testing that we can serialize and deserialize a class defined in the test class, with serialization/deserialization logic contained entirely in the test method.  What are we actually confirming at this point?

Also, seems like this test ought to be split in 2, the second when/then is just testing printing, it has nothing to do with serialization.  If this test failed, it wouldn't really pinpoint what is broken.

Reply:
I think the point is that the exceptions can be serialized with _java serialization_ which might not always serialize all relevant fields (I'm not an expert here, though there's certainly `transient` and I think you can override how an object is serialized by implementing some methods). So, this test would be checking whether this exception is explicitly supported by java serialization and doesn't throw away its child exceptions when serializing. 

As for the second part of the test, the same behavior is tested above in an above test. This seems to attempt to verify that even after serialization the printing behavior remains. Though, if we verify the first when/then, we should be pretty confident given that the above test works without needing to test the behavior again after de-serialization.

Reply:
I see your point on the java serialization, I suppose it is valuable to test that something in the class hierarchy hasn't broken the built-in serialization mechanism.  I didn't notice the second when/then repeats assertions tested elsewhere - seems very deleteable in that case.

Reply:
Deleted the second when/then